player: fix seeking during dead time

zmb3 opened the pull request:
This addresses a bug in the new player API where the player does not respond to seeking during large periods of "dead time." For example, if there is 30 seconds of dead time between two events, the player is effectively in a 30 second sleep. Attempting to change the position of playback during this 30-second period works, but is not observed until the sleep completes.

With this change, the sleep period is canceled when the user seeks to a new position.

Fixes #38560

Changelog: Improve the responsiveness of the session player during long periods of idle time.

zmb3 committed:
player: fix seeking during dead time

This addresses a bug in the new player API where the player does
not respond to seeking during large periods of "dead time."
For example, if there is 30 seconds of dead time between two events,
the player is effectively in a 30 second sleep. Attempting to change
the position of playback during this 30-second period works, but is
not observed until the sleep completes.

With this change, the sleep period is canceled when the user seeks
to a new position.

Fixes #38560

zmb3 commented on lib/player/player.go:
```diff
@@ -273,16 +275,26 @@ func (p *Player) SetPos(d time.Duration) error {
 		return errClosed
+	case <-p.wake:
+		return nil
 	case <-p.clock.After(time.Duration(scaled)):
```
I'm not particularly concerned. This happens at "human speed" and not in a hot loop.

zmb3 commented:
Looks like I have some flakiness to sort out 😞

kimlisa commented:
when testing, the idle time no longer freezes me, but trying to skip over the idle time, doesn't resume recording as it should:

https://www.loom.com/share/892305a678974f00ad293310b37d787f

the timer and bar is incrementing, but the recording doesn't show me anything (and if i let it, timer and bar keeps going after the end time)

zmb3 committed:
Improve UX when the user seeks while paused

Restart the stream in this case rather than trying to keep server
and client state in sync.

zmb3 committed:
Fix

zmb3 committed:
Update desktop player

zmb3 committed:
Lint/test fixes

zmb3 committed:
increase timeout

zmb3 committed:
Use a real clock for the new test
